src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Action/RestartAzureVMCommand.cs

```diff
@@ -65,6 +66,25 @@ public class RestartAzureVMCommand : VirtualMachineBaseCmdlet
         [Parameter(
            Mandatory = true,
            Position = 1,
+           ParameterSetName = RestartResourceGroupNameParameterSet,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "The virtual machine name.")]
+        [Parameter(
+           Mandatory = true,
+           Position = 1,
+           ParameterSetName = PerformMaintenanceResourceGroupNameParameterSet,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "The virtual machine name.")]
+        [Parameter(
+           Mandatory = false,
+           Position = 1,
+           ParameterSetName = RestartIdParameterSet,
```

Comment:
Why are we adding this parameter to the ID parameter sets?

Reply:
It was originally in all parameter sets, all this is doing is making it optional in id parameter sets (check out the help files)